### PR TITLE
update exechelper to keptnutils

### PR DESCRIPTION
**Pull request by adigeak**
Commit for #4068

### CODE REVIEW

**Comment by christian-kreuzberger-dtx** [comment]
Please sign your commit - see https://github.com/keptn/keptn/pull/4104/checks?check_run_id=2611082761

**Comment by adigeak** [comment]
Is there any issue?

**Comment by johannes-b** [comment]
@adigeak thanks for signing the commit!

@christian-kreuzberger-dtx can I ask you to finally review the code changes please.

File: cli/cmd/generate_support_archive.go [hunk 1]
```diff
@@ -442,96 +442,96 @@ func getIngressGateway(keptnNamespace string) *errorableStringResult {
...
 	fmt.Println("Retrieving namespaces")
-	writeErrorableStringResult(newErrorableStringResult(exechelper.ExecuteCommand("kubectl", "get namespaces")),
+	writeErrorableStringResult(newErrorableStringResult(keptnutils.ExecuteCommand("kubectl", []string{"get", "namespaces"})),
 		filepath.Join(dir, "namespaces.txt"))
 }
 
 func writeConfigMaps(namespace, dir string) {
 	fmt.Println("Retrieving list of config maps in " + namespace)
-	writeErrorableStringResult(newErrorableStringResult(exechelper.ExecuteCommand("kubectl", "get cm -n "+namespace)),
+	writeErrorableStringResult(newErrorableStringResult(keptnutils.ExecuteCommand("kubectl", []string{"get", "cm", "-n ", namespace})),
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/generate_support_archive.go:451]: Careful, when you set the parameter to `"-n "`, you end up getting an error message, as the command you generate contains an additional space between `-n` and the namespace, e.g.:
```
Error executing command kubectl get cm -n  keptn: exit status 1
Error from server (NotFound): namespaces " " not found
```

File: cli/cmd/generate_support_archive.go [hunk 2]
```diff
@@ -442,96 +442,96 @@ func getIngressGateway(keptnNamespace string) *errorableStringResult {
...
 		filepath.Join(dir, "configmap.txt"))
 }
 
 func writeSecrets(namespace, dir string) {
 	fmt.Println("Retrieving list of secrets in " + namespace)
-	writeErrorableStringResult(newErrorableStringResult(exechelper.ExecuteCommand("kubectl", "get secrets -n "+namespace)),
+	writeErrorableStringResult(newErrorableStringResult(keptnutils.ExecuteCommand("kubectl", []string{"get", "secrets", "-n ", namespace})),
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/generate_support_archive.go:457]: Please change `"-n "` to `"-n"`

File: cli/cmd/generate_support_archive.go [hunk 3]
```diff
@@ -442,96 +442,96 @@ func getIngressGateway(keptnNamespace string) *errorableStringResult {
...
 		filepath.Join(dir, "secrets.txt"))
 }
 
 func writeDeployments(namespace, dir string) {
 	fmt.Println("Retrieving list of deployments in " + namespace)
-	writeErrorableStringResult(newErrorableStringResult(exechelper.ExecuteCommand("kubectl", "get deployments -owide -n "+namespace)),
+	writeErrorableStringResult(newErrorableStringResult(keptnutils.ExecuteCommand("kubectl", []string{"get", "deployments", "-o", "wide", "-n ", namespace})),
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/generate_support_archive.go:463]: Please change `"-n "` to `"-n"`

File: cli/cmd/generate_support_archive.go [hunk 4]
```diff
@@ -442,96 +442,96 @@ func getIngressGateway(keptnNamespace string) *errorableStringResult {
...
 		filepath.Join(dir, "deployments.txt"))
 }
 
 func writePods(namespace, dir string) {
 	fmt.Println("Retrieving list of pods in " + namespace)
-	writeErrorableStringResult(newErrorableStringResult(exechelper.ExecuteCommand("kubectl", "get pods -owide  -n "+namespace)),
+	writeErrorableStringResult(newErrorableStringResult(keptnutils.ExecuteCommand("kubectl", []string{"get", "pods", "-o", "wide", "-n ", namespace})),
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/generate_support_archive.go:469]: Please change `"-n "` to `"-n"`

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
I found an issue when executing `keptn generate support-archive`, mainly caused by the `-n` Param.

**Comment by adigeak** [comment]
Done. Will take care of this next time.

File: cli/cmd/generate_support_archive.go [hunk 5]
```diff
@@ -442,96 +442,96 @@ func getIngressGateway(keptnNamespace string) *errorableStringResult {
...
 		return
 	}
 	for _, deployment := range strings.Split(strings.TrimSpace(res), "\n") {
-		res := newErrorableStringResult(exechelper.ExecuteCommand("kubectl",
-			"describe deployment "+deployment+" -n "+namespace))
+		res := newErrorableStringResult(keptnutils.ExecuteCommand("kubectl",
+			[]string{"describe", "deployment", deployment, "-n", namespace}))
 		writeErrorableStringResult(res, filepath.Join(dir, deployment+"_description.txt"))
 	}
 }
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/generate_support_archive.go:522]: I think this is not working.

`deploymentdescriptions.txt` contains the following output:

```
Error executing command kubectl get deployments --template {{range .items}}{{.metadata.name}}{{'
'}}{{end}} -n keptn: exit status 1
error: error parsing template {{range .items}}{{.metadata.name}}{{'
'}}{{end}}, template: output:1: unexpected unterminated character constant in command
```

In addition, I only see this in the global file `deploymentdescriptions.txt`, but I don't see the individual deployments.

File: cli/cmd/generate_support_archive.go [hunk 6]
```diff
@@ -442,96 +442,96 @@ func getIngressGateway(keptnNamespace string) *errorableStringResult {
...
 		return
 	}
 	for _, pod := range strings.Split(strings.TrimSpace(res), "\n") {
-		res := newErrorableStringResult(exechelper.ExecuteCommand("kubectl",
-			"describe pod "+pod+" -n "+namespace))
+		res := newErrorableStringResult(keptnutils.ExecuteCommand("kubectl",
+			[]string{"describe", "pod", pod, "-n", namespace}))
 		writeErrorableStringResult(res, filepath.Join(dir, pod+"_description.txt"))
 	}
 }
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/generate_support_archive.go:506]: I think this is also not working.

`poddescriptions.txt` contains the following output:

```
Error executing command kubectl get pods --template {{range .items}}{{.metadata.name}}{{'
'}}{{end}} -n keptn: exit status 1
error: error parsing template {{range .items}}{{.metadata.name}}{{'
'}}{{end}}, template: output:1: unexpected unterminated character constant in command
```

I only see this in the global file poddescriptions.txt, but I don't see the individual pods.

File: cli/cmd/generate_support_archive.go [hunk 7]
```diff
@@ -442,96 +442,96 @@ func getIngressGateway(keptnNamespace string) *errorableStringResult {
...
 		writeErrorableStringResult(newErrorableStringResult("", err),
 			filepath.Join(dir, "podlogs.txt"))
 		return
 	}
 	for _, pod := range strings.Split(strings.TrimSpace(res), "\n") {
-		res := newErrorableStringResult(exechelper.ExecuteCommand("kubectl", "logs "+pod+" --all-containers=true -n "+namespace))
+		res := newErrorableStringResult(keptnutils.ExecuteCommand("kubectl", []string{"logs", pod, "--all-containers=true", "-n", namespace}))
 		writeErrorableStringResult(res, filepath.Join(dir, pod+"_log.txt"))
 	}
 }
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/generate_support_archive.go:537]: Same here, `podlogs.txt` contains
```
Error executing command kubectl get pods --template {{range .items}}{{.metadata.name}}{{'
'}}{{end}} -n keptn: exit status 1
error: error parsing template {{range .items}}{{.metadata.name}}{{'
'}}{{end}}, template: output:1: unexpected unterminated character constant in command
```

No individual pod logs are available.

**Comment by christian-kreuzberger-dtx** [comment]
I will be on vacation the next couple of days. @bacherfl and @warber please have a look at this PR next week ;)

File: cli/cmd/generate_support_archive.go (hunk 5 above)
Comment (adigeak) [reply cli/cmd/generate_support_archive.go:522]: > res, err := keptnutils.ExecuteCommand("kubectl",
		[]string{"get", "deployments", "--template", `{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}`, "-n", namespace})
By using back tick we can make the "raw string literal.

**Comment by adigeak** [comment]
Sorry for the delay i was in the vaccination leave. @bacherfl and @warber please check the issue one back from weekend.

**Review by warber — APPROVED** [review]
tested creating a support archive locally, seems to work. lgtm

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM thanks

**Pull request merged**